webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/OrderedScrambles.java

```diff
@@ -0,0 +1,245 @@
+package net.gnehzr.tnoodle.server.webscrambles;
+
+import static net.gnehzr.tnoodle.utils.GsonUtils.GSON;
+import static net.gnehzr.tnoodle.utils.GwtSafeUtils.azzert;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.Days;
+
+import net.lingala.zip4j.exception.ZipException;
+import net.lingala.zip4j.io.ZipOutputStream;
+import net.lingala.zip4j.model.ZipParameters;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import com.itextpdf.text.DocumentException;
+
+public class OrderedScrambles {
+    
+    // TODO we are using Joda-Time for dates, via its .jar file
+    // We can update java to java8 and use java.time, which is the same class, as of JSR-310
+    // We can dismiss joda-time-2.10.1.jar from /lib
+
+    private static final String wcifIgnorableKey = "other";
+
+    public static void generateOrderedScrambles(String globalTitle, Date generationDate, ZipOutputStream zipOut, ZipParameters parameters, String scheduleJsonStringfied, String sheet) throws DocumentException, IOException, ZipException {
+        JsonObject scheduleJson = new JsonParser().parse(scheduleJsonStringfied).getAsJsonObject();
```

Comment:
I think this would be cleaner if we instead declare our own java class for the WCIF schedule, and let GSON parse the json into that for us. I think that will make some stuff nicer:

- integers will get convert to actual java integers (I see you're calling `Integer.parseInt` below)
- you should be able to access attributes via `.foo` rather than `.get` or `getAsJsonArray`

I think you can see an example of how to do this with how we parse the `ScrambleRequest` class.

Reply:
Done.